test/end2end_test.go

```diff
@@ -6730,6 +6730,73 @@ func testRPCTimeout(t *testing.T, e env) {
 	}
 }
 
+type FakeCompressor struct {
```

Comment:
No need to export.    It would be nice to parameterize the size (or add a payload buffer directly here?) for reusability.